pkg/kubelet/dockertools/manager.go

```diff
@@ -1202,7 +1202,9 @@ func (dm *DockerManager) killPodWithSyncResult(pod *api.Pod, runningPod kubecont
 	if networkContainer != nil {
 		ins, err := dm.client.InspectContainer(networkContainer.ID.ID)
 		if err != nil {
-			glog.Errorf("Error inspecting container %v: %v", networkContainer.ID.ID, err)
+			err = fmt.Errorf("Error inspecting container %v: %v", networkContainer.ID.ID, err)
+			glog.Error(err)
+			result.Fail(err)
```

Comment:
I think these all go down to the fact that `SyncPod` is a huge function that does millions things. The best way, IMO, to handle this is to separate out each action/func, and wrap each func with something like
```
    result := kubecontainer.NewSyncResult(kubecontainer.StartContainer, container.Name)
    defer func() {
        if err != nil {
            result.Fail(err, msg) 
        }           
    }()
    err, msg  = func() {
       // Actual work
       return kubecontainer.ErrRunContainer, err.Error()
    }
```